src/typescript-service.ts

```diff
@@ -100,25 +100,25 @@ export interface Settings extends PluginSettings {
 /**
  * Maps string-based CompletionEntry::kind to enum-based CompletionItemKind
  */
-const completionKinds: { [name: string]: CompletionItemKind } = {
-    class: CompletionItemKind.Class,
-    constructor: CompletionItemKind.Constructor,
-    enum: CompletionItemKind.Enum,
-    field: CompletionItemKind.Field,
-    file: CompletionItemKind.File,
-    function: CompletionItemKind.Function,
-    interface: CompletionItemKind.Interface,
-    keyword: CompletionItemKind.Keyword,
-    method: CompletionItemKind.Method,
-    module: CompletionItemKind.Module,
-    property: CompletionItemKind.Property,
-    reference: CompletionItemKind.Reference,
-    snippet: CompletionItemKind.Snippet,
-    text: CompletionItemKind.Text,
-    unit: CompletionItemKind.Unit,
-    value: CompletionItemKind.Value,
-    variable: CompletionItemKind.Variable,
-}
+const completionKinds = new Map<string, CompletionItemKind>([
+    [`class`, CompletionItemKind.Class],
+    [`constructor`, CompletionItemKind.Constructor],
+    [`enum`, CompletionItemKind.Enum],
+    [`field`, CompletionItemKind.Field],
+    [`file`, CompletionItemKind.File],
+    [`function`, CompletionItemKind.Function],
+    [`interface`, CompletionItemKind.Interface],
+    [`keyword`, CompletionItemKind.Keyword],
+    [`method`, CompletionItemKind.Method],
+    [`module`, CompletionItemKind.Module],
+    [`property`, CompletionItemKind.Property],
+    [`reference`, CompletionItemKind.Reference],
+    [`snippet`, CompletionItemKind.Snippet],
+    [`text`, CompletionItemKind.Text],
+    [`unit`, CompletionItemKind.Unit],
+    [`value`, CompletionItemKind.Value],
+    [`variable`, CompletionItemKind.Variable],
+])
```

Comment:
what was the reason to make this a Map, given it never changes?

Reply:
Using `constructor` as a object property name caused type system problems with `CompletionItemKind`, somehow it inferred that `Function` was a part of the union of numbers that `CompletionItemKind` is aliased to.  I couldn't find another way to escape the known keyword of `constructor` so using a map seemed to be the next best option.  Happy to change it back if you know of a better way!

Reply:
Ah, that makes total sense!